src/IPC/PipeRx.cc

```diff
@@ -1,22 +1,52 @@
-#include <iostream>
 #include "IPC.h"
 
+#include <iostream>
+#include <fcntl.h>      /* O_RDONLY */
+#include <unistd.h>     /* read() close() unlink() */
+#include <sys/stat.h>   /* mkfifo() */
+#include <errno.h>  
+#include <string.h>
+
+#define FIFO "./fifoChannel"
+
 void PipeRx::receive(void) {
     this->setupPipeRx();
 
-    this->fileSizeRx();
-
     this->pipeRx();
 }
 
 void PipeRx::setupPipeRx(void) {
-    ;
-}
-
-void PipeRx::fileSizeRx(void) {
-    ;
+    mkfifo(FIFO, 0666);
```

Comment:
I recommend actually to take the int that returns mkfifo in order to handle the error ( throw or whatever you want to do )
Because if your process interrupts you risk to have a pipe not deleted ( exemple : you press ctrl+c during the exec, the destructor isn't called and then your pipe still exist, for this you can check errno and if errno == EEXIST you delete it and re-create it.).

Reply:
I've added the following check:
![Screenshot from 2022-04-12 09-02-36](https://user-images.githubusercontent.com/97879338/162891402-8e2683e9-9cd2-4d25-8d3a-95a08143ea32.png)

I forgot to add the destructor functions, will add them as well.

Reply:
![Screenshot from 2022-04-13 07-37-59](https://user-images.githubusercontent.com/97879338/163101387-894b0686-8400-441f-a54c-13d1e88222c1.png) [Source](https://w3.cs.jmu.edu/kirkpams/OpenCSF/Books/csf/html/Pipes.html#fifos)

Don't think that's going to be a problem?

Reply:
Yes, but this is for launching it but if a process is killed I wonder if it will block during the transfer.